src/rehype-filter.js

```diff
@@ -4,6 +4,26 @@ const splice = [].splice
 
 module.exports = rehypeFilter
 
+/**
+ * @typedef {import('hast').Root} Root
+ * @typedef {import('hast').Element} Element
+ *
+ * @callback AllowElement
+ * @param {Element} element
+ * @param {number} index
+ * @param {Element|Root} parent
+ * @returns {boolean}
+ *
+ * @typedef {Object} RehypeFilterOptions
+ * @property {Array.<string>} [allowedElements]
```

Comment:
I personally find `string[]` to be a bit easier to read.
But am fine either way :+1:

Reply:
I tried both (in other projects). The shorter syntax is a bit nice. But I’m defaulting to JSDoc/Closure syntax (https://jsdoc.app/tags-type.html#overview) now in the hope of in the future using JSDoc or other tools to handle those annotations, where those tools might crash on TS syntax.